src/script/script.h

```diff
@@ -567,10 +567,14 @@ class CScript : public CScriptBase
      */
     unsigned int GetSigOpCount(const CScript& scriptSig) const;
 
+    // insightexplorer, there may be more script types in the future
+    enum ScriptType : int {
+        P2PKH = 1,
+        P2SH = 2,
+    };
     bool IsPayToPublicKeyHash() const;
     bool IsPayToScriptHash() const;
-
-    int Type() const;
+    boost::optional<ScriptType> GetType() const;
```

Comment:
This looks great! I wonder, could we change `enum ScriptType` to include `NONE = 0`. Doing this would allow us to change this to just return `ScriptType` which I think would simplify this slightly.

Reply:
Good suggestion, just pushed a new commit (will squash) to implement your suggestion.